run-tests.sh

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+cd "$(dirname "$0")"
+
+port=9000
+
+print_usage() {
+  cat <<EOF
+usage: $0 [IMAGE...]
+
+examples:
+       $0 "sonar-scanner-cli:4.1"
+EOF
+}
+
+info() {
+  echo "[info] $@"
+}
+
+warn() {
+  echo "[warn] $@" >&2
+}
+
+fatal() {
+  echo "[error] $@" >&2
+  exit 1
+}
+
+require() {
+  local prog missing=()
+  for prog; do
+    if ! type "$prog" &>/dev/null; then
+      missing+=("$prog")
+    fi
+  done
+
+  [[ ${#missing[@]} == 0 ]] || fatal "could not find reqired programs on the path: ${missing[@]}"
+}
+
+wait_for_sonarqube() {
+  local i web_up=no sonarqube_up=no
+
+  for ((i = 0; i < 10; i++)); do
+    info "waiting for web server to start ..."
+    if curl -sI localhost:$port | grep '^HTTP/.* 200'; then
+      web_up=yes
+      break
+    fi
+    sleep 5
+  done
+
+  [[ $web_up == yes ]] || return 1
+
+  for ((i = 0; i < 20; i++)); do
+    info "waiting for sonarqube to be ready ..."
+    if curl -s localhost:$port/api/system/status | grep '"status":"UP"'; then
+      sonarqube_up=yes
+      break
+    fi
+    sleep 10
+  done
+
+  [[ "$sonarqube_up" == yes ]]
+}
+
+test_scanner() {
```

Comment:
This function is called multiple times when several scanner-cli images are provided to the script

In this case, scanner could share:
1. the project clone
2. the scanner cache directory (`~/.sonar`)

`~/.sonar` is not a problem yet because this directory is not mounted into the image yet.
I wonder however, if the scanner is really considering the project as read-only. Maybe it is today, maybe it won't tomorrow.

Out of concern of predictability of the tests and reproducibility:
* should we rather make a new clone (maybe from an initial local clone) for each scanner image?
* not allow to run multiple images at one but ensure we have a new clone each time for each script call?

Reply:
For now the repo is cloned each time and destroyed after run.